### DISCUSSION TITLE
Using trame with Vue

### DISCUSSION THREAD

**Discussion by martinkulka** [Kitware/trame#147] [Q&A] [answered]
Hello,

I'm a university student woth a thesis in 3D medical imaging. I am not the best at web technologies so if I say something that doesn't make sense please forgive me.

I basically need to render NIfTI images online, all three planes of the scan and ideally a volume render of said image. I will also add a segmentation function using SimpleITK.

I wanted to ask, is there any point in using trame for my use case? I thought it would be cool to render and also modify the files in a trame instance with SimpleITK and send the rendered images through the network to a Vue app, where I could implement various widgets which could interact and send requests to the backend trame app, and maybe add some other functionality to the website.

I would really appreciate any ideas and suggestions about this use of said technologies, since I don't have any idea what the best solution could be.

Thank you.

**Comment by jourdain** [comment, marked as answer]
Hi @martinkulka ,

thanks for reaching out. I think trame could be indeed a good fit for that especially if web is not your forte.
With trame you should be able to create a Python app that bring VTK and ITK as dependency to load, process and visualize your data the way you want. 

Then with a simple vue structure directly available via Python you can control the various pieces you aim to dynamically edit. 

You can see some simple VTK example doing remote rendering here [1](https://github.com/Kitware/trame/blob/master/examples/06_vtk/01_SimpleCone/RemoteRendering.py), [2](https://github.com/Kitware/trame/blob/master/examples/06_vtk/02_ContourGeometry/RemoteRendering.py), [3](https://github.com/Kitware/trame-tutorial/blob/master/01_vtk/solution_flow.py), [4](https://github.com/Kitware/trame-tutorial/blob/master/01_vtk/solution_ray_cast.py) where 3 and 4 are direct translation of [vtk-examples](https://kitware.github.io/vtk-examples/site/)
